helpers/db.js

```diff
@@ -22,12 +22,19 @@ const productSchema = new Schema(
 
 const Product = models.Product || model("Product", productSchema);
 
+/**
+ * A async function to connect to a MongoDB capstone database
+ */
 async function connectDatabase() {
```

Comment:
First time I'm seeing that kind of function, we always used one that was longer and way more complicated. 
Could you send me a link to docs for this one, please?

Reply:
TL;DR:
```javascript
const Product = models.Product || model("Product", productSchema);
```
Creates a new model `model.Product` if it's not available
Otherwise reuses a existing model `model("Product", productSchema)`

---

Yes, the longer way is more complicated, since it uses the normal MongoDB methods.
This function is part of mongoose lib.
Just a warning, the documentation of mongoose is not really readable.
It took me a while to understand a bit of it.
[Models documentation](https://mongoosejs.com/docs/models.html)